ApplicationInsightsInternalLogger/src/main/java/com/microsoft/applicationinsights/internal/logger/InternalLogger.java

```diff
@@ -292,6 +357,9 @@ private void onInitializationError(String errorMessage) {
     }
 
     private void log(LoggingLevel requestLevel, String message, Object... args) {
+        if (!initialized && !systemPropertyInitialize()) {
```

Comment:
I would not recommend calling this method on every log call. Instead what you can do is that move this logic of initialization to, `TelemetryConfigurationFactory` class. It has the a method `setMinimumConfiguration()` if there is no config file. Also you can prioritize this initialization over config file if needed, but regardless the method should be called only once instead of calling it everytime on each log message.

Reply:
It's only called if not initialized.